src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -29,7 +43,7 @@ def consume(self, state):
         # hard we don't really need to deal with it right now.
         assert len(consumer.offsets) is 1
 
-        if state.HasField('stream_state'):
+        if state.stream_state.HasField('consumer_state'):
```

Comment:
> I guess `stream_state` would just be the default value now (no `consumer_state`) which is why this works?

I don't 100% follow this, but property access on protobuf messages never raises an `AttributeError`, which is why it's important to check `HasField` (which I'm pretty sure you already know).  In this conditional here we want to ensure that `state.stream_state.consumer_state` actually is a value and isn't just the default - hence the explicit `HasField` check.

Does that answer the question?